cmd/bnsd/x/statefix/msg.go

```diff
@@ -0,0 +1,26 @@
+package statefix
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/migration"
+)
+
+func init() {
+	migration.MustRegister(1, &ExecuteFixMsg{}, migration.NoModification)
+}
+
+var _ weave.Msg = (*ExecuteFixMsg)(nil)
+
+func (msg *ExecuteFixMsg) Validate() error {
+	var errs error
+	errs = errors.AppendField(errs, "Metadata", msg.Metadata.Validate())
+	if len(msg.FixID) == 0 {
```

Comment:
I think it is better to constrain the ID length. I propose 8 bytes would be appropriate.
```go
// ValidateSequence returns an error if this is not an 8-byte
// as expected for orm.IDGenBucket
func ValidateSequence(id []byte) error {
	if len(id) == 0 {
		return errors.Wrap(errors.ErrEmpty, "missing sequence")
	}
	if len(id) != 8 {
		return errors.Wrap(errors.ErrInput, "sequence is invalid length (expect 8 bytes)")
	}
	return nil
}
```
https://github.com/iov-one/weave/blob/master/orm/sequence.go#L68-L79

Reply:
`FixID` is not a sequence. It must be [something unique and declarable in code](https://github.com/iov-one/weave/pull/1028/files#diff-ca87377d4631b21143bf088e61c7d003R8). It is easier if this is a string.

My initial idea was to use a random string, generated offline (i.e. using `openssl rand -base64 8`). Such random string does not mean anything. 
We could use numbering instead, for example `0001` for the first one, `0002` for the second and so on. I think this is better than random string. It still lacks any valuable information. Sequence number information is not helpful.
My final approach is to use a string that is a short description. For exampe `"undo harm done by badboy"`. It is unique, reasonably short and it provides a good description of what it is about.

What do you think?

Reply:
IMHO `ExecuteFix` should contain some *description* that defines what the fix does. Best approach would be to have an auto-incremented ID field that defines the count of fixes out there and a `description` field